client/src/components/CharityDetalis/index.jsx

```diff
@@ -161,16 +164,34 @@ class CharityDetalis extends Component {
     const {
       tabs,
       charity: { name },
+      loading,
     } = this.state;
     const Contant = this.renderTab(tabs);
     return (
       <React.Fragment>
         <Header />
-        <div style={{ margin: '110px auto', width: '80%' }}>
-          <CharityHeader changeTab={this.changeTab} tabs={tabs} name={name} />
-          {Contant}
-        </div>
-        <Footer />
+        {loading ? (
+          <div className="loading-bubbles">
+            <ReactLoading
+              type="bubbles"
+              color="#f76009"
+              height="20%"
+              width="20%"
+            />
+          </div>
+        ) : (
+          <>
+            <div style={{ margin: '110px auto', width: '80%' }}>
+              <CharityHeader
+                changeTab={this.changeTab}
+                tabs={tabs}
+                name={name}
+              />
+              {Contant}
+            </div>
+            <Footer />
+          </>
+        )}
```

Comment:
@othman-shamla 
If this block, with the two large chunks of jsx conditional on `loading`, starts to get hard to work with, you could put each jsx block into a function and then just call those functions here in render():
`{loading ? this.renderLoadingBubbles() : this.renderContent()}`

At my work, we usually do that once we start to have really long conditional jsx like that, otherwise it starts to get hard to get all of the parenthesis and brackets correct.